src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -26,68 +26,70 @@
 
-        addToCart(request,response,product);
+        addToCart(request, response, product);
     }
 
-    private void addToCart(HttpServletRequest request, HttpServletResponse response,Product product) throws ServletException, IOException {
-        String error =null;
+    private void addToCart(HttpServletRequest request, HttpServletResponse response, Product product) throws ServletException, IOException {
+        final String ERROR = "error";
+        String error;
         try {
-            int quantity=getQuantity(request);
-            Cart cart=cartService.getCart(request);
+            int quantity = getQuantity(request);
+            Cart cart = cartService.getCart(request);
             cartService.add(cart, product, quantity);
-        }
-        catch (ParseException e){
-            error = "Not a number";
-        }
-        catch (OutOfStockException e){
-            error = "Not enough stock. Available "+e.getAvailableStock();
-        }
-        if (error!=null){
-            request.setAttribute("error",error);
-            showPage(request,response,product);
+        } catch (ParseException | OutOfStockException e) {
+            if (e instanceof ParseException) {
+                error = "Not a number";
+            } else {
+                error = "Not enough stock. Available " + ((OutOfStockException) e).getAvailableStock();
+            }
+            request.setAttribute(ERROR, error);
+            showPage(request, response, product);
             return;
         }
-        response.sendRedirect(request.getRequestURI()+"?success=true");
+        response.sendRedirect(request.getRequestURI() + "?success=true");
     }
 
     private int getQuantity(HttpServletRequest request) throws ParseException {
-        Locale locale=request.getLocale();
-        String quantityString = request.getParameter("quantity");
+        final String QUANTITY = "quantity";
+        Locale locale = request.getLocale();
+        String quantityString = request.getParameter(QUANTITY);
         return NumberFormat.getNumberInstance(locale).parse(quantityString).intValue();
     }
 
-    private long getProductID(HttpServletRequest request){
+    private long getProductID(HttpServletRequest request) {
         String uri = request.getRequestURI();
-        String productID = uri.substring(uri.lastIndexOf("/")+1);
-        return Long.valueOf(productID);
+        String productID = uri.substring(uri.lastIndexOf("/") + 1);
+        return Long.parseLong(productID);
     }
 
-    private void showPage(HttpServletRequest request,HttpServletResponse response,Product product) throws ServletException, IOException {
-        request.setAttribute("product", product);
-        request.setAttribute("cart",cartService.getCart(request));
+    private void showPage(HttpServletRequest request, HttpServletResponse response, Product product) throws ServletException, IOException {
+        final String PRODUCT = "product";
```

Comment:
Here are in other places: move constants to the top of the class. This way they can be reused later in other methods.